tensorflow/core/kernels/mkl/mkl_conv_ops.h

```diff
@@ -641,6 +641,9 @@ class MklConvBackpropCommonOp : public OpKernel {
     OP_REQUIRES(context, FormatFromString(data_format_str, &data_format_),
                 errors::InvalidArgument("Invalid data format"));
     OP_REQUIRES_OK(context, context->GetAttr("strides", &strides_));
+    if (strides_.size() < 4) {
```

Comment:
Shouldn't it be _exactly_ 4?

Reply:
If it's more the extra values are ignored, but that would be hard to debug. So let's pin to 4.

Reply:
> Shouldn't it be _exactly_ 4?

I am assuming that this same class used for Conv3D Ops like Conv3DBackpropInputV2 also. Isn't it ?

Reply:
Ah, I see.  But then it should probably match the data format size.  Though below, in the `data_format_`, it does seem to assume it is NCHW or NHWC.

Reply:
> Hence I think strides of Conv3D will be passed to there right? Correct me if i am wrong.

Okay, then don't put the check here.  Put it in the child class for Conv2d/3d.

> Will it validate this check implicitly or do we validate this explicitly ?

I believe so.

Reply:
> Okay, then don't put the check here. Put it in the child class for Conv2d/3d.
> 
> >

Both Conv2d/3d implements same class `MklConvCustomBackpropInputOp` . Hence it makes no difference I guess.

> > Will it validate this check implicitly or do we validate this explicitly ?
> 
> I believe so.

If strides validated at Register_Op itself then probably we can change the proposed check to validuate exactly for size=4.